### PR TITLE
folder reorganisation

### PR DESCRIPTION
**Pull request by PaLeroy**
TODO: check that everything still works 😆

### CODE REVIEW

**Comment by PaLeroy** [comment]
remaining todo: rewrite and check the neurips_23 paper folder to be "standalone" (-> everything executed from /blablabla/pymarl

**Comment by PaLeroy** [comment]
Almost all wrappers end with "Struct", do we keep that?

**Comment by PaLeroy** [comment]
I think the train and test example do not take campaign cost in consideration correctly

**Comment by PaLeroy** [comment]
Ready for the review :D

**Comment by moratodpg** [comment]
Excellent reorganization!!
Starting the review with very positive results:
image

File: README.md [hunk 1]
```diff
@@ -4,13 +4,14 @@
 **IMP-MARL** offers a platform for benchmarking the scalability of cooperative MARL methods in real-world engineering applications.
 
 In IMP-MARL, you can:
-- [Implement your own infrastructure management planning (IMP) environment or execute an available IMP environment](environments/).
-- [Train IMP policies through state-of-the-art MARL methods. The environments can be integrated with typical ecosystems via wrappers](imp_wrappers/).
-- [Compute expert-based heuristic policies](./heuristics/)
+- [Implement your own infrastructure management planning (IMP) environment or execute an available IMP environment](imp_marl/environments/).
+- [Train IMP policies through state-of-the-art MARL methods. The environments can be integrated with typical ecosystems via wrappers](imp_marl/imp_wrappers/).
+- [Compute expert-based heuristic policies](papers/neurips_23/heuristics/)
```
Comment (moratodpg) [inline README.md:9]: We could define a template class and incorporate it into imp_marl (in the future)

**Review by moratodpg — APPROVED** [review]
So far, I do not see anything problematic. I will try to reproduce the paper's experiments now.

**Comment by moratodpg** [comment]
> Almost all wrappers end with "Struct", do we keep that?

From my side, no problem to keep it as is for the moment :)

**Comment by moratodpg** [comment]
> I think the train and test example do not take campaign cost in consideration correctly

Can you please point to the specific place where we might have this problem?

File: papers/neurips_23/pymarl/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+## Retrieve seeds
+
+The seeds required to reproduce our paper are obtained by downloading the data from our experiments.
+
+These logs are available in [/resutls_scritps](imp_marl/papers/neurips_23/results_scripts/).
+
+Once downloaded, the [find_seed notebook](imp_marl/papers/neurips_23/results_scripts/find_seed.ipynb) allows you to retrieve seeds for a given configuration. 
+ 
+## Train the agents
+As example, we want to reproduce the results of QMIX in the k-out-of-n environment with 5 agents and correlations and campaign cost.
```
Comment (PaLeroy) [inline papers/neurips_23/pymarl/README.md:119]: I believe is was here @moratodpg

File: papers/neurips_23/pymarl/README.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ You need to install the packages from this repository.
 The `-e` allows you to modify imp_marl packages and use them without reinstalling it.
 
 ```
-pip install -e ../../../ .
+pip install -e ../../../
```
Comment (moratodpg) [inline papers/neurips_23/pymarl/README.md:49]: It now works fine

File: papers/neurips_23/pymarl/run_test.sh [hunk 4]
```diff
@@ -13,11 +13,13 @@ fi
...
 path=${exp_dir}${exp_name}
 n_test=$2 # number of test episodes
+campaign_option=$3
+seed_number=$4
 n_env=2 # number of environments to run in parallel
 name=test_${n_test}_${exp_name}
 echo $alg
 echo $env
-python pymarl_test.py --config=${alg} --env-config=${env} with test_nepisode=${n_test} checkpoint_path=${path} runner=parallel batch_size_run=${n_env} use_cuda=True name=${name}
+python pymarl_test.py --config=${alg} --env-config=${env} with test_nepisode=${n_test} checkpoint_path=${path} runner=parallel batch_size_run=${n_env} use_cuda=True name=${name} seed=${seed_number} env_args.campaign_cost=${campaign_option}
```
Comment (moratodpg) [inline papers/neurips_23/pymarl/run_test.sh]: I think I got your point. This addition should probably fix it.

File: papers/neurips_23/pymarl/README.md (hunk 3 above)
Comment (PaLeroy) [reply papers/neurips_23/pymarl/README.md:49]: Did you add and remove the dot again? :p

Comment (PaLeroy) [reply papers/neurips_23/pymarl/README.md:49]: See this commit https://github.com/moratodpg/imp_marl/pull/74/commits/47642ac0e8ac5e76dc88ca8002d29f32ca9d5102

**Review by PaLeroy — COMMENTED** [review]
I guess I am okay with this big merge :D

File: papers/neurips_23/pymarl/README.md [hunk 5]
```diff
@@ -46,7 +46,7 @@ You need to install the packages from this repository.
 The `-e` allows you to modify imp_marl packages and use them without reinstalling it.
 
 ```
-pip install -e ../../../ 
+pip install -e ../../../ .
```
Comment (moratodpg) [inline papers/neurips_23/pymarl/README.md]: It should be without 'dot' - I tested it xD

File: papers/neurips_23/pymarl/README.md (hunk 3 above)
Comment (moratodpg) [reply papers/neurips_23/pymarl/README.md:49]: Yeah, I checked it, and it works well without the dot. We are already providing a directory as argument.

File: papers/neurips_23/pymarl/README.md (hunk 5 above)
Comment (PaLeroy) [reply papers/neurips_23/pymarl/README.md]: 😆

**Pull request merged**